llvm/test/tools/sycl-post-link/virtual-functions/properties.ll

```diff
@@ -0,0 +1,42 @@
+; RUN: sycl-post-link -split=auto -properties -S < %s -o %t.table
+;
+; Device images with virtual functions in them should have
+; "virtual-functions-set" property under corresponding property set.
+; Device images that use virtual functions should have
+; "uses-virtual-functions-set" property under corresponding property set.
+; There can't be a device image where both properties are present, they are
+; mutually exclusive
+
+; RUN: FileCheck %s --input-file=%t_0.ll --check-prefix=CHECK-IR0
+; RUN: FileCheck %s --input-file=%t_1.ll --check-prefix=CHECK-IR1
+; RUN: FileCheck %s --input-file=%t_0.prop --check-prefix=CHECK-PROPS0
+; RUN: FileCheck %s --input-file=%t_1.prop --check-prefix=CHECK-PROPS1
+
+; CHECK-IR0: define spir_func void @foo
+; CHECK-IR1: define weak_odr dso_local spir_kernel void @kernel
+; CHECK-PROPS0: [SYCL/virtual functions]
+; CHECK-PROPS0-NEXT: virtual-functions-set
```

Comment:
there should also be the value of the `virtual-functions-set` property right? should we verify that, as well as the same for the `uses-virtual-functions-set` side?

Reply:
The fact that we use base64 encoding in properties makes such checks unreadable, i.e. there is little guarantee that whatever I put in there indeed correspond to strings we wanted to see there.

However, it will at least check that there are no unexpected changes to what we emit, so it is still useful, I will add those

Reply:
Test improved in 07c1ca60a65afdd4c0c57d36999da1ec3fef98d7